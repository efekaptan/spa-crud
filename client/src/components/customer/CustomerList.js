import React from 'react';
import Paging from 'reactjs-paging';
import { pageSize } from '../../configuration';
import { Link } from 'react-router-dom';

const CustomerList = ({ match, customers, pageIndex, totalCount, onPageClick, onDeleteClick }) =>
    <div className="row">
        <div className="col-lg-6">
            <div className="card">
                <div className="card-header">
                    <i className="fa fa-align-justify"></i> <strong>Customers</strong> List
                </div>
                <div className="card-block">
                    <table className="table table-bordered table-striped">
                        <thead>
                            <tr>
                                <th>Id</th>
                                <th>First Name</th>
                                <th>Last Name</th>
                                <th>Actions</th>
                            </tr>
                        </thead>
                        <tbody>
                            {customers.map(customer =>
                                <tr key={customer.id}>
                                    <td>{customer.id}</td>
                                    <td>{customer.firstName}</td>
                                    <td>{customer.lastName}</td>
                                    <td className="buttons">
                                        <Link className="btn btn-warning btn-sm" to={`/add-edit-customer/${customer.id}`}>Edit</Link>
                                        <button className="btn btn-danger btn-sm" onClick={() => onDeleteClick(customer.id)}>Delete</button>
                                    </td>
                                </tr>
                            )}
                        </tbody>
                    </table>
                    <div className="row">
                        <div className="col-lg-9">
                            <nav>
                                {(totalCount > pageSize) &&
                                    <Paging
                                        pageIndex={pageIndex}
                                        groupSize={5}
                                        navSize={2}
                                        totalCount={totalCount}
                                        pageSize={pageSize}
                                        onClick={onPageClick} />
                                }
                            </nav>
                        </div>
                        <div className="col-lg-3">
                            <Link to={`/add-edit-customer`} className="btn btn-primary">Create New</Link>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    </div>

export default CustomerList;