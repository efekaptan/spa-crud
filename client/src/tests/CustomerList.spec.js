import React from 'react'
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import CustomerList from '../components/customer/CustomerList';
import getContextProvider, { contextObject } from './ContextProvider';

configure({ adapter: new Adapter() });

const mockCustomers = [{ "id": 15, "firstName": "Eliseo", "lastName": "Goldfarb", "birthday": "1969-11-19T21:00:00.000Z", "gender": "m", "lastContact": "2017-03-18T12:20:06.702Z", "customerLifetimeValue": 1019.91 }, { "id": 14, "firstName": "Iva", "lastName": "Buenrostro", "birthday": "1979-06-30", "gender": "w", "lastContact": "2017-01-02-29T21:08:50.700Z", "customerLifetimeValue": 213.12 }, { "id": 13, "firstName": "Fredrick", "lastName": "Bellah", "birthday": "1991-02-21", "gender": "m", "lastContact": "2017-08-01T11:57:47.142Z", "customerLifetimeValue": 0 }, { "id": 12, "firstName": "Willena", "lastName": "Condrey", "birthday": "1987-05-03", "gender": "w", "lastContact": "2017-07-08T13:18:56.888Z", "customerLifetimeValue": 50.99 }, { "id": 11, "firstName": "Alexis", "lastName": "Kuss", "birthday": "1996-10-12", "gender": "m", "lastContact": "2017-06-01T23:28:56.782Z", "customerLifetimeValue": 191.12 }, { "id": 10, "firstName": "Broderick", "lastName": "Longnecker", "birthday": "1969-11-21", "gender": "m", "lastContact": "2017-03-18T12:20:06.702Z", "customerLifetimeValue": 1019.91 }, { "id": 9, "firstName": "Chanda", "lastName": "Copenhaver", "birthday": "1979-06-30", "gender": "w", "lastContact": "2017-01-02-29T21:08:50.700Z", "customerLifetimeValue": 213.12 }, { "id": 8, "firstName": "Nicky", "lastName": "Slay", "birthday": "1991-02-21", "gender": "m", "lastContact": "2017-08-01T11:57:47.142Z", "customerLifetimeValue": 0 }, { "id": 7, "firstName": "Karoline", "lastName": "Aultman", "birthday": "1987-05-03", "gender": "w", "lastContact": "2017-07-08T13:18:56.888Z", "customerLifetimeValue": 50.99 }, { "id": 6, "firstName": "Clayton", "lastName": "Desousa", "birthday": "1996-10-12", "gender": "m", "lastContact": "2017-06-01T23:28:56.782Z", "customerLifetimeValue": 191.12 }];
const totalCount = 15;

describe('Customer List Snapshot', () => {
    const reactElement = getContextProvider(contextObject,
        <CustomerList customers={mockCustomers} pageIndex="1" totalCount={totalCount} />
    );
    it('Capturing Snapshot of Customer List Snapshot', () => {
        const renderedValue = renderer.create(reactElement).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});

describe('Component Render Tests', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(
            <MemoryRouter>
                <CustomerList customers={mockCustomers} pageIndex="1" totalCount={totalCount} />
            </MemoryRouter>);
    });

    it('Render the DUMB component', () => {
        expect(wrapper.length).toEqual(1);
    });

    it('Contains 11 rows', () => {
        expect(wrapper.find('tr').length).toEqual(11);
    });

    it('First row, first name is = "Eliseo"', () => {
        expect(wrapper.find('tr').at(1).find('td').at(1).text()).toEqual('Eliseo');
    });
});