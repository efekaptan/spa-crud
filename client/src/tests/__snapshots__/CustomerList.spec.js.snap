// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer List Snapshot Capturing Snapshot of Customer List Snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-6"
  >
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        <i
          className="fa fa-align-justify"
        />
         
        <strong>
          Customers
        </strong>
         List
      </div>
      <div
        className="card-block"
      >
        <table
          className="table table-bordered table-striped"
        >
          <thead>
            <tr>
              <th>
                Id
              </th>
              <th>
                First Name
              </th>
              <th>
                Last Name
              </th>
              <th>
                Actions
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                15
              </td>
              <td>
                Eliseo
              </td>
              <td>
                Goldfarb
              </td>
              <td
                className="buttons"
              >
                <a
                  className="btn btn-warning btn-sm"
                  href={undefined}
                  onClick={[Function]}
                >
                  Edit
                </a>
                <button
                  className="btn btn-danger btn-sm"
                  onClick={[Function]}
                >
                  Delete
                </button>
              </td>
            </tr>
            <tr>
              <td>
                14
              </td>
              <td>
                Iva
              </td>
              <td>
                Buenrostro
              </td>
              <td
                className="buttons"
              >
                <a
                  className="btn btn-warning btn-sm"
                  href={undefined}
                  onClick={[Function]}
                >
                  Edit
                </a>
                <button
                  className="btn btn-danger btn-sm"
                  onClick={[Function]}
                >
                  Delete
                </button>
              </td>
            </tr>
            <tr>
              <td>
                13
              </td>
              <td>
                Fredrick
              </td>
              <td>
                Bellah
              </td>
              <td
                className="buttons"
              >
                <a
                  className="btn btn-warning btn-sm"
                  href={undefined}
                  onClick={[Function]}
                >
                  Edit
                </a>
                <button
                  className="btn btn-danger btn-sm"
                  onClick={[Function]}
                >
                  Delete
                </button>
              </td>
            </tr>
            <tr>
              <td>
                12
              </td>
              <td>
                Willena
              </td>
              <td>
                Condrey
              </td>
              <td
                className="buttons"
              >
                <a
                  className="btn btn-warning btn-sm"
                  href={undefined}
                  onClick={[Function]}
                >
                  Edit
                </a>
                <button
                  className="btn btn-danger btn-sm"
                  onClick={[Function]}
                >
                  Delete
                </button>
              </td>
            </tr>
            <tr>
              <td>
                11
              </td>
              <td>
                Alexis
              </td>
              <td>
                Kuss
              </td>
              <td
                className="buttons"
              >
                <a
                  className="btn btn-warning btn-sm"
                  href={undefined}
                  onClick={[Function]}
                >
                  Edit
                </a>
                <button
                  className="btn btn-danger btn-sm"
                  onClick={[Function]}
                >
                  Delete
                </button>
              </td>
            </tr>
            <tr>
              <td>
                10
              </td>
              <td>
                Broderick
              </td>
              <td>
                Longnecker
              </td>
              <td
                className="buttons"
              >
                <a
                  className="btn btn-warning btn-sm"
                  href={undefined}
                  onClick={[Function]}
                >
                  Edit
                </a>
                <button
                  className="btn btn-danger btn-sm"
                  onClick={[Function]}
                >
                  Delete
                </button>
              </td>
            </tr>
            <tr>
              <td>
                9
              </td>
              <td>
                Chanda
              </td>
              <td>
                Copenhaver
              </td>
              <td
                className="buttons"
              >
                <a
                  className="btn btn-warning btn-sm"
                  href={undefined}
                  onClick={[Function]}
                >
                  Edit
                </a>
                <button
                  className="btn btn-danger btn-sm"
                  onClick={[Function]}
                >
                  Delete
                </button>
              </td>
            </tr>
            <tr>
              <td>
                8
              </td>
              <td>
                Nicky
              </td>
              <td>
                Slay
              </td>
              <td
                className="buttons"
              >
                <a
                  className="btn btn-warning btn-sm"
                  href={undefined}
                  onClick={[Function]}
                >
                  Edit
                </a>
                <button
                  className="btn btn-danger btn-sm"
                  onClick={[Function]}
                >
                  Delete
                </button>
              </td>
            </tr>
            <tr>
              <td>
                7
              </td>
              <td>
                Karoline
              </td>
              <td>
                Aultman
              </td>
              <td
                className="buttons"
              >
                <a
                  className="btn btn-warning btn-sm"
                  href={undefined}
                  onClick={[Function]}
                >
                  Edit
                </a>
                <button
                  className="btn btn-danger btn-sm"
                  onClick={[Function]}
                >
                  Delete
                </button>
              </td>
            </tr>
            <tr>
              <td>
                6
              </td>
              <td>
                Clayton
              </td>
              <td>
                Desousa
              </td>
              <td
                className="buttons"
              >
                <a
                  className="btn btn-warning btn-sm"
                  href={undefined}
                  onClick={[Function]}
                >
                  Edit
                </a>
                <button
                  className="btn btn-danger btn-sm"
                  onClick={[Function]}
                >
                  Delete
                </button>
              </td>
            </tr>
          </tbody>
        </table>
        <div
          className="row"
        >
          <div
            className="col-md-9 col-sm-9"
          >
            <nav>
              <ul
                className="pagination"
              >
                <li
                  className="page-item active"
                >
                  <a
                    onClick={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                      }
                    }
                  >
                    1
                  </a>
                </li>
                <li
                  className="page-itemnull"
                >
                  <a
                    onClick={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                      }
                    }
                  >
                    2
                  </a>
                </li>
              </ul>
            </nav>
          </div>
          <div
            className="col-md-3 col-sm-3"
          >
            <a
              className="btn btn-primary"
              href={undefined}
              onClick={[Function]}
            >
              Create New
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
